packages/react-jss/src/createUseStyles.js

```diff
@@ -42,111 +35,75 @@ const createUseStyles = <Theme: {}>(styles: Styles<Theme>, options?: HookOptions
     const context = React.useContext(JssContext)
     const theme = useTheme()
 
-    const [state, dispatch] = React.useReducer(reducer, null, () => {
-      const sheet = createStyleSheet({
-        context,
-        styles,
-        name,
-        theme,
-        index,
-        sheetOptions
-      })
-
-      let dynamicRules
-      let classes
-      if (sheet) {
-        if (context.registry) {
-          context.registry.add(sheet)
-        }
-        dynamicRules = addDynamicRules(sheet, data)
-        classes = getSheetClasses(sheet, dynamicRules)
-      }
-
-      return {
-        sheet,
-        dynamicRules,
-        classes: classes || {}
-      }
-    })
-
-    useEffectOrLayoutEffect(
+    const [sheet, dynamicRules] = React.useMemo(
```

Comment:
If I remember correctly there was a problem with using `React.useMemo` as it can be rerun by React at a given point of time.

I think @eps1lon knew more about this.

Reply:
I think you are talking about the theoretical situation in async mode, not sure if it was made real.